homeassistant/components/devolo_home_network/device_tracker.py

```diff
@@ -0,0 +1,162 @@
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONNECTED_STATIONS,
+    CONNECTED_WIFI_CLIENTS,
+    DOMAIN,
+    MAC_ADDRESS,
+    WIFI_APTYPE,
+    WIFI_BANDS,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and sensors and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+    coordinators: dict[str, DataUpdateCoordinator] = hass.data[DOMAIN][entry.entry_id][
+        "coordinators"
+    ]
+    registry = entity_registry.async_get(hass)
+    tracked = set()
+
+    @callback
+    def new_device_callback() -> None:
+        """Add new devices if needed."""
+        new_entities = []
+        for station in coordinators[CONNECTED_WIFI_CLIENTS].data[CONNECTED_STATIONS]:
+            if station[MAC_ADDRESS] in tracked:
+                continue
+
+            new_entities.append(
+                DevoloScannerEntity(
+                    coordinators[CONNECTED_WIFI_CLIENTS], device, station[MAC_ADDRESS]
+                )
+            )
+            tracked.add(station[MAC_ADDRESS])
+            if new_entities:
+                async_add_entities(new_entities)
+
+    @callback
+    def restore_entities() -> None:
+        """Restore clients that are not a part of active clients list."""
+        missing = []
+        for entity in registry.entities.values():
+            mac_address = entity.unique_id.replace(f"{device.serial_number}_", "")
+            if (
+                entity.config_entry_id == entry.entry_id
+                and entity.platform == DOMAIN
+                and entity.domain == DEVICE_TRACKER_DOMAIN
+                and mac_address not in tracked
+            ):
+                missing.append(
+                    DevoloScannerEntity(
+                        coordinators[CONNECTED_WIFI_CLIENTS], device, mac_address
+                    )
+                )
+                tracked.add(mac_address)
+
+        if missing:
+            async_add_entities(missing)
+
+    if device.device and "wifi1" in device.device.features:
+        restore_entities()
+        entry.async_on_unload(
+            coordinators[CONNECTED_WIFI_CLIENTS].async_add_listener(new_device_callback)
+        )
+
+
+class DevoloScannerEntity(CoordinatorEntity, ScannerEntity):
+    """Representation of a devolo device tracker."""
+
+    def __init__(
+        self, coordinator: DataUpdateCoordinator, device: Device, mac: str
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator)
+        self._device = device
+        self._mac = mac
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str]:
+        """Return the attributes."""
+        attrs: dict[str, str] = {}
+        if self.coordinator.data[CONNECTED_STATIONS]:
+            station: dict[str, Any] = next(
```

Comment:
Cache if it's a station once. This will never change right?

Reply:
Well, a station could do e.g. band steering. So rarely the wifi band could change.